crypto/s2n_aead_cipher_aes_gcm.c

```diff
@@ -389,29 +389,49 @@ static S2N_RESULT s2n_aead_cipher_aes128_gcm_set_ktls_info(struct s2n_ktls_crypt
 
     RESULT_ENSURE_LTE(sizeof(crypto_info->key), in->key.size);
     RESULT_CHECKED_MEMCPY(crypto_info->key, in->key.data, sizeof(crypto_info->key));
-
-    RESULT_ENSURE_LTE(sizeof(crypto_info->iv), in->iv.size);
-    RESULT_CHECKED_MEMCPY(crypto_info->iv, in->iv.data, sizeof(crypto_info->iv));
-
     RESULT_ENSURE_LTE(sizeof(crypto_info->rec_seq), in->seq.size);
     RESULT_CHECKED_MEMCPY(crypto_info->rec_seq, in->seq.data, sizeof(crypto_info->rec_seq));
 
-    /* The salt is a prefix of the IV
+    /* TLS1.2 uses partially explicit nonces. That means that although part of the
+     * nonce is still fixed and implicit (the salt), the remainder is explicit
+     * (written into the record) and must be unique per record. The RFC5288 suggests
+     * using the sequence number, which is how s2n-tls implements its nonces.
+     *
+     * Therefore, ktls expects the salt to contain the iv derived from the secret
+     * and should generate the remainder of the nonce per-record.
      *
-     *= https://www.rfc-editor.org/rfc/rfc4106#section-4
-     *# The salt field is a four-octet value that is assigned at the
-     *# beginning of the security association, and then remains constant
-     *# for the life of the security association.
+     * See the TLS1.2 RFC:
+     * - https://datatracker.ietf.org/doc/html/rfc5246#section-6.2.3.3
+     * And RFC5288, which defines the TLS1.2 AES-GCM cipher suites:
+     * - https://datatracker.ietf.org/doc/html/rfc5288#section-3
      */
     RESULT_ENSURE_LTE(sizeof(crypto_info->salt), in->iv.size);
     RESULT_CHECKED_MEMCPY(crypto_info->salt, in->iv.data, sizeof(crypto_info->salt));
 
+    /* Because TLS1.2 uses partially explicit nonces, the kernel should not
+     * make use of the iv in crypto_info but instead use the current sequence number,
+     * updated with each record sent.
```

Comment:
The IV can be any non-repeating value, because it's explicit. Unlike TLS1.3, it doesn't HAVE to be the sequence number. I removed that part of the comment.